rendermodules/intensity_correction/calculate_multiplicative_correction.py

```diff
@@ -58,48 +96,69 @@ def run(self):
             # used for easy creation of tilespecs for output stack
             firstts.append(tilespecs[0])
             numtiles += len(tilespecs)
-
-        #subsample in the case where the number of tiles is too large
+        # subsample in the case where the number of tiles is too large
         if self.args['num_images'] > 0:
-            alltilespecs = randomly_subsample_tilespecs(alltilespecs,self.args['num_images'])
-
-        # read images and create stack
-        N, M, img0 = getImage(alltilespecs[0])
-        stack = np.zeros((N, M, numtiles), dtype=img0.dtype)
-        mypartial = partial(getImageFromTilespecs, alltilespecs)
-        indexes = range(0, numtiles)
-        with renderapi.client.WithPool(self.args['pool_size']) as pool:
-            images = pool.map(mypartial, indexes)
-
-        # calculate median
-        for i in range(0, len(images)):
-            stack[:, :, i] = images[i]
-        np.median(stack, axis=2, overwrite_input=True)
-        (A, B, C) = stack.shape
-        if (numtiles % 2 == 0):
-            med1 = stack[:, :, numtiles // 2 - 1]
-            med2 = stack[:, :, numtiles // 2 + 1]
-            med = (med1 + med2) // 2
-        else:
-            med = stack[:, :, (numtiles - 1) // 2]
-        med = gaussian_filter(med, 10)
-
-        # save image and create output tilespecs
+            alltilespecs = randomly_subsample_tilespecs(
+                alltilespecs, self.args['num_images'])
+
+        stackInfo = renderapi.stack.get_full_stack_metadata(self.args['input_stack'],
+                                                            render=self.render)
+        channel_names = stackInfo['stats'].get('channelNames', [])
+        # Quick fix to have the empty channel logic work when channelNames=[]
+        if len(channel_names) == 0:     
+            channel_names = [None]
+
+        #figure out which of our channels is the default channel
+        if channel_names[0] is not None:
+            try:
+                default_chan = next(ch.name for ch in firstts[0].channels if ch.ip[0].imageUrl == firstts[0].ip[0].imageUrl )
+            except StopIteration:
+                raise RenderModuleException("default channel doesn't match any channel")
+
         outdir = self.args['output_directory']
         if not os.path.exists(outdir):
             os.makedirs(outdir)
+        print("channel_name",channel_names)
```

Comment:
This should be able to go to the argschemaparser logger

Reply:
This doesn't need to exist, logger or not. Removing.